test/actor_test.py

```diff
@@ -924,28 +924,57 @@ def test_actor_multiple_gpus_from_multiple_tasks(shutdown_only):
         redirect_output=True,
         num_cpus=(num_local_schedulers * [10 * num_gpus_per_scheduler]),
         num_gpus=(num_local_schedulers * [num_gpus_per_scheduler]))
+    time.sleep(1)
```

Comment:
@stephanie-wang @ujvl I just looked at this test, and the doesn't make any sense. It really shouldn't have been passing before (except it probably passed because the timeout in the line `ready_ids, _ = ray.wait([a.get_location_and_ids.remote()], timeout=10)` was so short.

But when a `create_actors` task finishes, it's actors should go out of scope and so all the GPUs will get freed up again.

I'm not sure if this test is testing anything useful. Maybe we should just remove it?

Reply:
Hmm yeah I'm not sure if the test is written correctly, but currently I'm trying to debug performance; the test seems to hang for a long time **inside** `create_actors` (similar to #3088 ?) before it even gets to the asserts.

Reply:
Interesting. It's trying to create 100 actors, so there's going to be some rate-limiting based on the number of CPUs.

If it's related to this PR, then we should fix it of course, but if not, then maybe we should get this merged and look into it separately?

Reply:
Yeah I'm just unsure if this PR is exacerbating some existing problem or creating a new one, the test passes on `master` but not on this branch.

On master the test finishes immediately whereas on this branch there are > `50s` gaps between actor creations, even if I reduce the number of actors.

Reply:
Hmm yeah, it might make sense to separate out the fix for the unit test into its own PR. I think the PR was passing on `master`, but I think it was also significantly slower than without xray. Can't remember for sure right now.